### ISSUE TITLE
new lesson: Introduction to Strings

### ISSUE THREAD

**Issue by Utkarsh1504** [Utkarsh1504/DSA-Java#25]
### ⭐ Request Type

Add in-depth documentation on the Strings in Java.
The markdown file is located [here](https://github.com/Utkarsh1504/DSA-Java/blob/main/lessons/intro-string.md).

### Lesson Content
The lesson must contain at least these are the topics:

1. Introduction
2. What is String?
3. String Creation: String Literal
4. String Creation: new Keyword
4. How Strings Stored In Memory
5. String Pooling
6. String Immutability In Java
7. String Comparison
8. PrintStream Class
9. Important Strings Methods
10. String Operations
10. Format Specifier
11. String VS String Buffer
12. Some code examples
13. Conclusion

### 💻 Location

This lesson is present under the
Section: `6`
Lesson: `A`
Location Order: `6A`
File-Name: [`intro-string.md`](https://github.com/Utkarsh1504/DSA-Java/blob/main/lessons/intro-string.md)

### ❌ Additional Info

You can take the help of other markdown files in the [/lesson](https://github.com/Utkarsh1504/DSA-Java/blob/main/lessons/loops.md) directory so that you get the idea of in which fashion you should write.

If you don't know how to do markdown, don't worry it is quite easy, just follow this docs-https://guides.github.com/features/mastering-markdown/

If your concepts are not that good follow this Insertion sort video of @kunal-kushwaha (https://www.youtube.com/watch?v=zL1DPZ0Ovlo)

**Comment by aniket-24** [comment]
Please assign me this issue!

**Comment by Utkarsh1504** [comment]
@aniket-24 this will be quite lengthy and needs a lot of documentation. 
Take your time and make contributions, Thank you

**Comment by Mansivjain00** [comment]
Hey i would like to work on this issue too!

If aniket is doing it too, maybe we could divide the work. Also i would like to do this under the hacktoberfest

**Comment by Utkarsh1504** [comment]
Yes, absolutely you can, as this lesson is going to be quite long you can divide the work. actually, I'm raising another issue for the same, doing the half syllabus for each one

1. Introduction
2. What is String?
3. String Creation: String Literal
4. String Creation: new Keyword
5. How Strings Stored In Memory
6. String Pooling
7. String Immutability In Java
8. String Comparison


9. PrintStream Class
10. Important Strings Methods
11. String Operations
12. Format Specifier
13. String VS String Buffer
14. Some code examples
15. Conclusion

**Comment by Utkarsh1504** [comment]
@Mansivjain00 drop a comment on the second issue I'll assign you
